src/core_plugins/kibana/public/visualize/editor/visualization_editor.js

```diff
@@ -17,36 +17,36 @@
  * under the License.
  */
 
-import './visualize.less';
-import './visualize_legend';
-import { uiModules } from '../modules';
+import { uiModules } from 'ui/modules';
 import 'angular-sanitize';
-import { VisEditorTypesRegistryProvider } from '../registry/vis_editor_types';
-import { getUpdateStatus } from '../vis/update_status';
+import { VisEditorTypesRegistryProvider } from 'ui/registry/vis_editor_types';
 
 uiModules
   .get('kibana/directive', ['ngSanitize'])
-  .directive('visualizationEditor', function (Private, $timeout) {
+  .directive('visualizationEditor', function (Private, $timeout, getAppState) {
     const editorTypes = Private(VisEditorTypesRegistryProvider);
 
     return {
       restrict: 'E',
       scope: {
-        vis: '=',
-        visData: '=',
+        savedObj: '=',
         uiState: '=?',
-        searchSource: '='
+        timeRange: '='
       },
       link: function ($scope, element) {
       // Clone the _vis instance.
-        const vis = $scope.vis;
+        const vis = $scope.savedObj.vis;
         const Editor = typeof vis.type.editor === 'function' ? vis.type.editor :
           editorTypes.find(editor => editor.key === vis.type.editor);
         const editor = new Editor(element[0], vis);
 
         $scope.renderFunction = () => {
-          if (!$scope.vis) return;
-          editor.render($scope.visData, $scope.searchSource, getUpdateStatus(Editor.requiresUpdateStatus, $scope, $scope), $scope.uiState);
+          if (!vis) return;
```

Comment:
I know it already has been in there before, but since we are anyway doing major refactoring, I think we should also think about some of that old code: Why is this here? How could it ever happen the savedObj is defined but the `vis` on it isn't? Also - and that's new due to this PR -  if it isn't defined initially, it will never be defined afterwards, since we'll never jump back to the top of the linking function where we actually get the `vis` reference (beforehand it could also be defined afterwards).

Reply:
i think this can safely be removed, i'll do some testing to verify